core/src/rust/src/index.rs

```diff
@@ -0,0 +1,129 @@
+//! Methods to create / destroy the index
+
+use jni::{
+    objects::{JClass, JObjectArray, JString},
+    sys::jlong,
+    JNIEnv,
+};
+use tantivy::{
+    directory::MmapDirectory,
+    schema::{
+        FacetOptions, Field, JsonObjectOptions, Schema, SchemaBuilder, TextFieldIndexing, FAST,
+        INDEXED, STORED, STRING,
+    },
+    Index, TantivyDocument,
+};
+
+use crate::{
+    errors::{JavaException, JavaResult},
+    exec::jni_exec,
+    jnienv::JNIEnvExt,
+    state::{
+        field_constants::{self, facet_field_name, LABEL_LIST},
+        IndexHandle,
+    },
+};
+
+pub const WRITER_MEM_BUDGET: usize = 50 * 1024 * 1024;
```

Comment:
Suppose we run this in a container, I believe this comes from the total - heap mem? Do you think there is value in making this configurable or this is a good enough statically configured value?

Reply:
My general thought is that this is already smaller than Lucene, so I'm not concerned about it causing new issues.  If we think there's legit reason to change to a config value it's easy enough to do, but my gut says to do that once we have a need to change the value to avoid config explosion.